src/org/zaproxy/zap/extension/pscanrules/FingerprintingHeadersScanner.java

```diff
@@ -0,0 +1,118 @@
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+
+/**
+ * a scanner to passively scan for the presence of the server HTTP response header
+ */
+public class FingerprintingHeadersScanner extends PluginPassiveScanner {
+
+    /**
+     * Prefix for internationalised messages used by this rule
+     */
+    private static final String MESSAGE_PREFIX = "pscanrules.fingerprintingheaderscanner.";
+
+    private PassiveScanThread parent = null;
+
+    private final List<String> fingerprintingHeaders = new ArrayList<String>();
+
+    public FingerprintingHeadersScanner() {
+        fingerprintingHeaders.add("Server");
+        fingerprintingHeaders.add("X-Powered-By");
+        fingerprintingHeaders.add("X-AspNet-Version");
+        fingerprintingHeaders.add("X-AspNetMvc-Version");
+    }
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        if (AlertThreshold.LOW.equals(this.getLevel())) {
+            return;
+        }
+        boolean headerPresent = false;
+        List<String> fingerprintingHeadersFound = new ArrayList<String>();
+
+        for (String header :
+                fingerprintingHeaders) {
+            Vector<String> found = msg.getResponseHeader().getHeaders(header);
+
+            if (found != null){
+                fingerprintingHeadersFound.add(header + ":" + found.firstElement());
+            }
+        }
+
+        if (fingerprintingHeadersFound.isEmpty()) {
+            return;
+        }
+
+        this.raiseAlert(msg, id, String.join(",", fingerprintingHeadersFound) );
+    }
+
+    private void raiseAlert(HttpMessage msg, int id, String foundFingerprintingHeaders) {
+        Alert alert = new Alert(getPluginId(), Alert.RISK_LOW, Alert.CONFIDENCE_HIGH,  getName());
+        alert.setDetail(
+                Constant.messages.getString(MESSAGE_PREFIX + "desc"),
+                msg.getRequestHeader().getURI().toString(),
+                "Headers", //parameter
+                "", 					//attack
+                Constant.messages.getString(MESSAGE_PREFIX + "extrainfo"),		//other info
+                Constant.messages.getString(MESSAGE_PREFIX + "soln"),			//solution
+                Constant.messages.getString(MESSAGE_PREFIX + "refs"),			//refs
+                (foundFingerprintingHeaders), 	//evidence, if any
+                933, //CWE-933: OWASP Top Ten 2013 Category A5 - Security Misconfiguration
+                14,  //WASC-14: Server Misconfiguration
+                msg);
+
+        parent.raiseAlert(id, alert);
+    }
+
+    @Override
+    public int getPluginId() {
+        return 1078;
```

Comment:
This ID should be reserved (in https://github.com/zaproxy/zaproxy/blob/develop/src/doc/scanners.md).